pkg/cloud/amazon/vault/vault_resources.go

```diff
@@ -370,29 +449,68 @@ func runCloudformationTemplate(templateBody *string, stackName *string) error {
 		Capabilities: []*string{
 			aws.String("CAPABILITY_NAMED_IAM"),
 		},
+		Parameters: parameters,
 	}
 
 	_, err := svc.CreateStack(input)
 	if err != nil {
-		return errors.Wrapf(err, "Unable to create vault preresiquite resources")
+		return errors.Wrapf(err, "Unable to create vault prerequisite resources")
 	}
-
+	donec := make(chan bool)
+	go runStackEventsReporting(donec, svc, stackName)
 	// Wait until stack is created
 	err = svc.WaitUntilStackCreateComplete(desInput)
 	if err != nil {
-		return errors.Wrapf(err, "Unable to create vault preresiquite resources")
+		return errors.Wrapf(err, "Unable to create vault prerequisite resources")
 	}
-
-	log.Logger().Debugf("Ran the vault cloudformation template successfully")
+	donec <- true
+	log.Logger().Debugf("Ran the vault CloudFormation template successfully")
 	return nil
 }
 
+func extractVaultStackOutputs(stackName string) (*StackOutputs, error) {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	// Create CloudFormation client in region
+	svc := cloudformation.New(sess)
+
+	desInput := &cloudformation.DescribeStacksInput{StackName: aws.String(stackName)}
+
+	output, err := svc.DescribeStacks(desInput)
+	if err != nil {
+		return nil, err
+	}
+	vsp := &StackOutputs{}
+	if len(output.Stacks) > 0 {
+		stack := output.Stacks[0]
+		for _, v := range stack.Outputs {
+			switch *v.OutputKey {
+			case "AWSS3Bucket":
+				vsp.S3BucketARN = v.OutputValue
+			case "AWSKMSKey":
+				vsp.KMSKeyARN = v.OutputValue
+			case "AWSDynamoDBTable":
+				vsp.DynamoDBTableARN = v.OutputValue
+			default:
+				log.Logger().Warnf("CloudFormation parameter %s not expected", *v.OutputKey)
+			}
+		}
+	}
```

Comment:
I would be nice to validate at this point that none of the expected `vsp` fields is empty. For instance, an output value gets missed or is empty.

Reply:
I'm sure the CloudFormation stack would fail if it can't produce the expected outputs, because it would mean that the creation of the resource that it's getting the reference from would've failed.